packages/@divvi/mobile/src/viem/index.ts

```diff
@@ -1,53 +1,83 @@
 
 export const viemTransports: Record<Network, Transport> = {
-  [Network.Celo]: http(),
-  [Network.Ethereum]: http(networkConfig.alchemyRpcUrl[Network.Ethereum], {
-    fetchOptions: {
-      headers: {
-        Authorization: `Bearer ${ALCHEMY_ETHEREUM_API_KEY}`,
-      },
-    },
-  }),
-  [Network.Arbitrum]: http(networkConfig.alchemyRpcUrl[Network.Arbitrum], {
-    fetchOptions: {
-      headers: {
-        Authorization: `Bearer ${ALCHEMY_ARBITRUM_API_KEY}`,
-      },
-    },
-  }),
-  [Network.Optimism]: http(networkConfig.alchemyRpcUrl[Network.Optimism], {
-    fetchOptions: {
-      headers: {
-        Authorization: `Bearer ${ALCHEMY_OPTIMISM_API_KEY}`,
-      },
-    },
-  }),
-  [Network.PolygonPoS]: http(networkConfig.alchemyRpcUrl[Network.PolygonPoS], {
-    fetchOptions: {
-      headers: {
-        Authorization: `Bearer ${ALCHEMY_POLYGON_POS_API_KEY}`,
-      },
-    },
-  }),
-  [Network.Base]: http(networkConfig.alchemyRpcUrl[Network.Base], {
-    fetchOptions: {
-      headers: {
-        Authorization: `Bearer ${ALCHEMY_BASE_API_KEY}`,
-      },
-    },
-  }),
+  [Network.Celo]: http(
+    ALCHEMY_API_KEY ? networkConfig.alchemyRpcUrl[Network.Celo] : undefined,
+    ALCHEMY_API_KEY
+      ? {
+          fetchOptions: {
+            headers: {
+              Authorization: `Bearer ${ALCHEMY_API_KEY}`,
+            },
+          },
+        }
+      : {}
+  ),
+  [Network.Ethereum]: http(
+    ALCHEMY_API_KEY ? networkConfig.alchemyRpcUrl[Network.Ethereum] : undefined,
+    ALCHEMY_API_KEY
+      ? {
+          fetchOptions: {
+            headers: {
+              Authorization: `Bearer ${ALCHEMY_API_KEY}`,
+            },
+          },
+        }
+      : {}
+  ),
+  [Network.Arbitrum]: http(
+    ALCHEMY_API_KEY ? networkConfig.alchemyRpcUrl[Network.Arbitrum] : undefined,
+    ALCHEMY_API_KEY
+      ? {
+          fetchOptions: {
+            headers: {
+              Authorization: `Bearer ${ALCHEMY_API_KEY}`,
+            },
+          },
+        }
+      : {}
+  ),
+  [Network.Optimism]: http(
+    ALCHEMY_API_KEY ? networkConfig.alchemyRpcUrl[Network.Optimism] : undefined,
+    ALCHEMY_API_KEY
+      ? {
+          fetchOptions: {
+            headers: {
+              Authorization: `Bearer ${ALCHEMY_API_KEY}`,
+            },
+          },
+        }
+      : {}
+  ),
+  [Network.PolygonPoS]: http(
+    ALCHEMY_API_KEY ? networkConfig.alchemyRpcUrl[Network.PolygonPoS] : undefined,
+    ALCHEMY_API_KEY
+      ? {
+          fetchOptions: {
+            headers: {
+              Authorization: `Bearer ${ALCHEMY_API_KEY}`,
+            },
+          },
+        }
+      : {}
+  ),
+  [Network.Base]: http(
+    ALCHEMY_API_KEY ? networkConfig.alchemyRpcUrl[Network.Base] : undefined,
+    ALCHEMY_API_KEY
+      ? {
+          fetchOptions: {
+            headers: {
+              Authorization: `Bearer ${ALCHEMY_API_KEY}`,
+            },
+          },
+        }
+      : {}
+  ),
```

Comment:
Since we're using the same Alchemy Key could we shorten it to something like the code below?

```TypeScript
const SUPPORTED_NETWORKS = [
  Network.Celo,
  Network.Ethereum,
  Network.Arbitrum,
  Network.Optimism,
  Network.PolygonPoS,
  Network.Base,
] as const

const alchemyOptions = ALCHEMY_API_KEY
  ? {
      fetchOptions: {
        headers: { Authorization: `Bearer ${ALCHEMY_API_KEY}` },
      },
    }
  : undefined

const makeTransport = (n: Network) =>
  http(ALCHEMY_API_KEY ? networkConfig.alchemyRpcUrl[n] : undefined, alchemyOptions)

export const viemTransports = Object.fromEntries(
  SUPPORTED_NETWORKS.map((n) => [n, makeTransport(n)])
) satisfies Record<Network, Transport>
```